niceman/distributions/debian.py

```diff
@@ -179,6 +194,30 @@ def normalize(self):
         #   would make us require supporting flexible typing -- string or a list
         pass
 
+    def satisfies_package(self, package):
+        """return True if this distribution (self) satisfies the requirements 
+        of the passed package"""
+        if not isinstance(package, Package):
+            raise TypeError('satisfies_package() requires a package argument')
+        if not isinstance(package, DEBPackage):
+            return False
+        for p in self.packages:
+            if p.satisfies(package):
+                return True
+        return False
+
+    def satisfies(self, other):
+        """return True if this distribution (self) satisfies the requirements 
+        of the other distribution (other)"""
+        if not isinstance(other, Distribution):
+            raise TypeError('satisfies() requires a package argument')
+        if not isinstance(other, DebianDistribution):
+            return False
+        for p in other.packages:
+            if not self.satisfies_package(p):
+                return False
+        return True
```

Comment:
above four lines could be more pythonicly expressed
```python
return all(map(self.satisfies_package, other.packages))
```
in python3 map is already a generator (so you would get early termination), in python2 -- not yet so it would go through all possibly for no reason.  So just 
```python
from six.moves import map
```